actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserConsentActor.java

```diff
@@ -63,19 +86,81 @@ private void getUserConsent(Request request) {
     }
 
     private void updateUserConsent(Request request) {
+        Map<String, Object> consent = (Map<String, Object>) request.getRequest().getOrDefault(JsonKey.CONSENT, new HashMap<String, Object>());
+        String userId = (String) consent.getOrDefault(JsonKey.USER_ID, "");
+        if(StringUtils.isEmpty(userId)){
+            userId = (String) request.getContext().get(JsonKey.REQUESTED_BY);
+        }
+        String managedFor = (String)request.getContext().get(JsonKey.MANAGED_FOR);
+        if(StringUtils.isNotEmpty(managedFor)){
+            userId = managedFor;
+        }
+        consent.put(JsonKey.USER_ID,userId);
+        userService.validateUserId(request, null, request.getRequestContext());
+
+        Map consentReq = createConsentRequest(consent, request.getRequestContext());
+
+        userConsentService.updateConsent(consentReq, request.getRequestContext());
+
         Response response = new Response();
-        Map<String, Object> consent = (Map<String, Object>) request.getRequest().getOrDefault("consent", new HashMap<String, Object>());
-        String userId = (String) consent.getOrDefault("userId", "");
-        String consumerId = (String) consent.getOrDefault("consumerId", "");
-        String objectId = (String) consent.getOrDefault("objectId", "");
-        consentCache.put(getKey(userId, consumerId, objectId), consent);
-        response.put("consent", new HashMap<String, Object>() {{ put("userId", userId);}});
-        response.put("message", "User Consent updated successfully.");
+        Map consentResponse = new HashMap<String, Object>();
+        consentResponse.put(JsonKey.USER_ID, userId);
+        response.put(JsonKey.CONSENT, consentResponse);
+        response.put(JsonKey.MESSAGE, "User Consent updated successfully.");
+
         sender().tell(response, self());
+
+        Map<String, Object> targetObject = null;
+        List<Map<String, Object>> correlatedObject = new ArrayList<>();
+        targetObject =
+                TelemetryUtil.generateTargetObject(
+                        userId, TelemetryEnvKey.USER, JsonKey.UPDATE, null);
+        TelemetryUtil.telemetryProcessingCall(
+                consent, targetObject, correlatedObject, request.getContext());
     }
 
     private String getKey (String userId, String consumerId, String objectId) {
         return "usr-consent:" + userId + ":" + consumerId + ":" + objectId;
     }
 
+    private Map<String, Object> createConsentRequest(Map<String, Object> consent, RequestContext context){
```

Comment:
this is possibly constructing a consent map, the term create is a little off the mark here.

Reply:
Done!